go/ephemeral/user_ek.go

```diff
@@ -52,29 +55,61 @@ func postNewUserEK(ctx context.Context, g *libkb.GlobalContext, sig string, boxe
 	return err
 }
 
-func PublishNewUserEK(ctx context.Context, g *libkb.GlobalContext) (data keybase1.UserEkMetadata, err error) {
+func PublishNewUserEK(ctx context.Context, g *libkb.GlobalContext) (metadata keybase1.UserEkMetadata, err error) {
 	defer g.CTrace(ctx, "PublishNewUserEK", func() error { return err })()
 
 	currentMerkleRoot, err := g.GetMerkleClient().FetchRootFromServer(ctx, libkb.EphemeralKeyMerkleFreshness)
 	if err != nil {
-		return data, err
+		return metadata, err
 	}
 
-	// TODO: Read the actual generation from the userEK store.
-	var generation keybase1.EkGeneration = 1
-
 	seed, err := newUserEphemeralSeed()
 	if err != nil {
-		return data, err
+		return metadata, err
 	}
 
-	// TODO: Store the key.
+	storage := g.GetUserEKBoxStorage()
+	generation, err := storage.MaxGeneration(ctx)
+	if err != nil {
+		// Let's try to get the max from the server
```

Comment:
Given that we're going to retry below anyway if the server rejects this push, should we just modify `MaxGeneration` to return `0` if there's nothing in storage? Could that save us some of complexity here?

Reply:
`MaxGeneration` should already return a 0 and not an error if storage is empty. so we need to try both i think if we get the storage error or we get rejected for a bad value in storage